cirq-google/cirq_google/engine/engine_sampler.py

```diff
@@ -139,3 +139,106 @@ def get_engine_sampler(
         )
     gate_set = gate_sets.NAMED_GATESETS[gate_set_name]
     return engine.get_engine(project_id).sampler(processor_id=processor_id, gate_set=gate_set)
+
+
+def get_device_sampler(
```

Comment:
I think we need a more bespoke name for this function. `get_device_sampler` is too generic, whereas this function has a very specific purpose and likely won't be used by general users in their scripts or applications. It's mainly to make our examples work in a variety of settings: (in colab, locally) x (real device, simulator)

Maybe something like: `get_engine_sampler_or_simulator`,
`try_authenticate_and_get_device`, `get_qcs_objects_for_notebook`, `get_engine_objects_or_mock`, ... idk does anyone else have any suggestions